test/images/image-util.sh

```diff
@@ -117,23 +144,50 @@ push() {
   docker_version_check
   TAG=$(<"${IMAGE}"/VERSION)
   if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+    os_archs=$(listOsArchs)
+    # NOTE(claudiub): if the REMOTE_DOCKER_URL var is not set, or it is an empty string, we must skip
+    # pushing the Windows image and including it into the manifest list.
+    if [[ ((! -v "REMOTE_DOCKER_URL") || -z "${REMOTE_DOCKER_URL}") && -n "$(printf "%s\n" $os_archs | grep '^windows')" ]]; then
+      echo "Skipping pushing the image '${IMAGE}' for Windows. REMOTE_DOCKER_URL should be set, containing the URL to a Windows docker daemon."
+      os_archs=$(printf "%s\n" $os_archs | grep -v "^windows")
+    fi
   else
-    archs=${!QEMUARCHS[@]}
+    # prepend linux/ to the QEMUARCHS items.
+    os_archs=$(printf 'linux/%s\n' "${!QEMUARCHS[@]}")
   fi
-  for arch in ${archs}; do
-    docker push "${REGISTRY}/${IMAGE}-${arch}:${TAG}"
+  for os_arch in ${os_archs}; do
+    if [[ $os_arch =~ .*/.* ]]; then
+      os_name=$(echo $os_arch | cut -d "/" -f 1)
+      arch=$(echo $os_arch | cut -d "/" -f 2)
+    else
+      echo "The BASEIMAGE file for the ${IMAGE} image is not properly formatted. Expected entries to start with 'os/arch', found '${os_arch}' instead."
+      exit 1
+    fi
+
+    if [[ "$os_name" = "linux" ]]; then
+      docker push "${REGISTRY}/${IMAGE}:${TAG}-${os_name}-${arch}"
+    else
+      # NOTE(claudiub): We're pushing the image we built on the remote Windows node.
+      docker --tlsverify -H "${REMOTE_DOCKER_URL}" push "${REGISTRY}/${IMAGE}:${TAG}-${os_name}-${arch}"
```

Comment:
I think we want to gate this behind the same check you're using for `build` above, ie:
```sh
elif [[ -v "REMOTE_DOCKER_URL" && ! -z "${REMOTE_DOCKER_URL}" ]]; then
#...
else
  echo "Cannot push ..."
fi
```

Reply:
Sure, but it shouldn't be the case in the first place. At L150, if ``$REMOTE_DOCKER_URL`` is not set, we filter out the ``windows/`` entries from ``os_archs``, so the ``elif`` branch will never be executed.